cpp/dolfin/io/XDMFFile.cpp

```diff
@@ -69,15 +69,11 @@ std::string rank_to_string(std::size_t value_rank)
 // Returns true for DG0 function::Functions
 bool has_cell_centred_data(const function::Function& u)
 {
-  int cell_based_dim = 1;
-  for (int i = 0; i < u.value_rank(); i++)
-    cell_based_dim *= u.function_space()->mesh()->topology().dim();
-
   assert(u.function_space());
-  assert(u.function_space()->dofmap());
-  assert(u.function_space()->dofmap()->element_dof_layout);
-  return (u.function_space()->dofmap()->element_dof_layout->num_dofs()
-          == cell_based_dim);
+  return (u.function_space()->element()->family().compare(
```

Comment:
Signatures are for user convenience and shouldn't be be relied upon internally (this caused a lot of pain in the past). `degree` should also be avoided.

Comparing `ElementDofLayout::num_entity_dofs(tdim)` and `ElementDofLayout::num_dofs()` should work.